jupyter-lib/shared-compiler/src/main/kotlin/org/jetbrains/kotlinx/jupyter/debug/DebugIntegration.kt

```diff
@@ -0,0 +1,21 @@
+package org.jetbrains.kotlinx.jupyter.debug
+
+import kotlin.concurrent.thread
+
+abstract class DebugIntegrationProvider(
+    protected val debugPort: Int?,
+) {
+    companion object {
+        const val DEBUG_THREAD_NAME = "NotebookDebugThread"
+    }
+
+    fun provideDebugPort(): Int? = debugPort
+
+    fun provideSuspendedThread(runnable: () -> Unit): Thread {
```

Comment:
This abstraction looks leaking. I would suggest to have just a boolean flag here and create the thread in place it's needed